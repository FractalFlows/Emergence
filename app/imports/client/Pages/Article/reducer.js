/*
 * Built by Astrocoders
 * @flow
 */

import {
  flow,
  set,
} from 'lodash/fp'

const initialState = {
  article: {},
}

type State = {
  article: {},
}

type Action = {
  type: string,
  payload: { [key: string]: any },
}

export default function inspections(state: State = initialState, { type, payload }: Action): State {
  switch (type) {
    default:
      return state
  }
}
