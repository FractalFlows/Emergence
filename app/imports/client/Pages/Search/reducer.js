import {
  FETCH_SEARCH,
  LOAD_SEARCH_RESULTS,
  CLEAR_SEARCH,
  FETCH_SELECTED_ARTICLE,
  LOAD_SELECTED_ARTICLE,
} from './constants'

type Article = {
  slug: string,
  DOI: string,
  abstract: string,
  authors: Array<string>,
}

type State = {
  articlesFromSearch: Array<Article>,
  searchText: string,
  isSearch: bool,
}

const initialState = {
  articlesFromSearch: [],
  searchText: '',
  isSearching: false,
}

export default function search(state: State = initialState, { type, payload }: Action){
  switch(type){
    case FETCH_SEARCH: 
      return {
        ...state,
        isSearching: true,
        searchText: payload.searchText,
      }
    case LOAD_SEARCH_RESULTS:
      return {
        ...state,
        isSearching: false,
        articlesFromSearch: payload.articles,
      }
    case CLEAR_SEARCH: 
      return {
        ...state,
        searchText: '',
        articlesFromSearch: [],
        isSearching: false,
      }
    case FETCH_SELECTED_ARTICLE: 
      return {
        ...state,
        isFetchingSelectedArticle: true,
      }
    case FETCH_SELECTED_ARTICLE: 
      return {
        ...state,
        isFetchingSelectedArticle: true,
      }
    case LOAD_SELECTED_ARTICLE: 
      return {
        ...state,
        isFetchingSelectedArticle: false,
      }
    default: return { ...state }
  }
} 
