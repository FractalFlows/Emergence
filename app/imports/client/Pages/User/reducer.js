/*
 * Built by Astrocoders
 * @flow
 */

const initialState = {

}

type State = {
}

type Action = {
  type: string,
  payload: any,
}

export default function user(state:State = initialState, { type, payload }:Action):State {
  switch (type) {
    default:
      return state
  }
}
