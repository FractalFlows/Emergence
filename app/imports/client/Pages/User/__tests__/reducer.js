import {
  loginInit,
  loginCheck,
  loginError,
  loginSuccess,
  loginClean,
  refreshToken,
  userFetchData,
} from '../actions'
import reducer from '../reducer'

describe('User reducer', () => {
  it('should handle USER_LOGIN_INIT properly', () => {
    const state =  {
      token: false,
      error: false,
      mutationErrors: [],
      loading: false,
    }
    const action = loginInit()

    expect(reducer(state, action)).toMatchSnapshot()
  })

  it('should handle USER_LOGIN_SUCCESS properly', () => {
    const state =  {
      token: false,
      error: false,
      mutationErrors: [],
      loading: true,
    }
    const action = loginSuccess({
      token: 'astrocoders_rocks_and_this_is_not_a_secret',
    })

    expect(reducer(state, action)).toMatchSnapshot()
  })

  it('should handle USER_LOGIN_CLEAN properly', () => {
    const state =  {
      token: false,
      error: true,
      mutationErrors: [],
      loading: true,
    }
    const action = loginClean()

    expect(reducer(state, action)).toMatchSnapshot()
  })

  it('should handle USER_REFRESH_TOKEN properly', () => {
    const state =  {
      token: 'one_token_right_here',
      error: false,
      mutationErrors: [],
      loading: false,
    }
    const action = refreshToken({
      token: 'one_new_token_right_here_also',
    })

    expect(reducer(state, action)).toMatchSnapshot()
  })

  it('should handle USER_LOGIN_CHECK properly', () => {
    const state =  {
      token: 'one_token_right_here',
      error: false,
      mutationErrors: [],
      loading: false,
    }
    const action = loginCheck({
      routeName: 'some_neat_route',
    })

    expect(reducer(state, action)).toMatchSnapshot()
  })

  it('should handle USER_FETCH_DATA properly', () => {
    const state =  {
      token: false,
      error: false,
      mutationErrors: [],
      loading: false,
    }
    const action = userFetchData({
      data: {
        id: 'x',
        name: 'Edward Elric',
        level: 'alchemist',
      }
    })

    expect(reducer(state, action)).toMatchSnapshot()
  })
})
