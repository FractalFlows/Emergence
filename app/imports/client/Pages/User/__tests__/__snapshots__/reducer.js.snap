exports[`User reducer should handle USER_FETCH_DATA properly 1`] = `
Object {
  "error": false,
  "loading": false,
  "me": Object {
    "id": "x",
    "level": "alchemist",
    "name": "Edward Elric",
  },
  "mutationErrors": Array [],
  "token": false,
}
`;

exports[`User reducer should handle USER_LOGIN_CHECK properly 1`] = `
Object {
  "error": false,
  "loading": true,
  "mutationErrors": Array [],
  "routeName": "some_neat_route",
  "token": "one_token_right_here",
}
`;

exports[`User reducer should handle USER_LOGIN_CLEAN properly 1`] = `
Object {
  "error": false,
  "loading": false,
  "mutationErrors": Array [],
  "token": false,
}
`;

exports[`User reducer should handle USER_LOGIN_INIT properly 1`] = `
Object {
  "error": false,
  "loading": true,
  "mutationErrors": Array [],
  "token": false,
}
`;

exports[`User reducer should handle USER_LOGIN_SUCCESS properly 1`] = `
Object {
  "error": false,
  "loading": false,
  "mutationErrors": Array [],
  "token": "astrocoders_rocks_and_this_is_not_a_secret",
}
`;

exports[`User reducer should handle USER_REFRESH_TOKEN properly 1`] = `
Object {
  "error": false,
  "loading": false,
  "mutationErrors": Array [],
  "token": "one_new_token_right_here_also",
}
`;
